gremlin-python/pom.xml

```diff
@@ -83,102 +83,31 @@ limitations under the License.
-                                    </target>
-                                </configuration>
-                            </execution>
-                            <!-- copy files in python directory to target/py and run virtual env to sandbox python.
-                                 there is no need to "activate" the virtualenv because all calls to python occur
-                                 directly from bin/ -->
-                            <execution>
-                                <id>setup-py-env</id>
+                                <id>setup-env</id>
                                 <phase>process-resources</phase>
                                 <goals>
                                     <goal>run</goal>
                                 </goals>
                                 <configuration>
                                     <tasks>
-                                        <!-- seems like we need a few different copies of the same source. all the
-                                             different python stuff doesn't seem to want to share. we use
-                                             /python3 for basic tests and stuff and /python-packaged for distribution
-                                             tasks. some of the problem seems to stem from the python lifecycle not
-                                             binding perfectly well to the maven lifecycle (integration tests seems to
-                                             cause troubles specifically). note that the commands to install wheel are
-                                             largely for safety in case someone is using an older version of virtualenv
-                                             (which doesn't install wheel by default) -->
+                                        <!-- pytest will create python-reports inside target/python3/python-reports
+                                        as docker compose cannot access outside its working directory -->
                                         <copy todir="${project.build.directory}/python3">
                                             <fileset dir="src/main/python"/>
                                         </copy>
-                                        <exec dir="${project.build.directory}/python3" executable="python3"
-                                              failonerror="true">
-                                            <arg line="--version"/>
-                                        </exec>
-                                        <exec dir="${project.build.directory}/python3" executable="virtualenv"
-                                              failonerror="true">
-                                            <arg line="--version"/>
-                                        </exec>
-                                        <exec dir="${project.build.directory}/python3" executable="virtualenv"
```

Comment:
as i sit here admiring all this pom.xml code getting simplified behind docker, i wonder if these changes will allow releases to actually work. for example, here we get rid of the virtualenv setup of python3, wheel, setuptools, etc. which would all be needed on release day (i think) when we deploy to pypi with `-Dpypi`: https://github.com/apache/tinkerpop/pull/1797/files#diff-924caaf2e60b71e4f805fe7d35f9c8895cb8d23ae258b12b7497b21d91ab639fR249-R269

thoughts?

Reply:
That is one thing I wasn't sure of, how much the impact would be to release, not only with the set up removed, but also with the packaging taken inside of docker. I don't have any experiences with `pypi` releases though, I don't imagine there is any `mock release` to try? In any case, we could move the necessary python environment set up into the release profile to make sure it still works.

Some other options could be to move the packaging stage for python back into maven, so we keep the environment set up, to be sure it works the same way as before for package and release. Then down the road we could consider putting both package and release into Docker.

Reply:
> In any case, we could move the necessary python environment set up into the release profile to make sure it still works.

i was thinking you should just do this for now. in the slightly longer term (i.e. next release manager tries this approach out and documents it), maybe we just go to native release instructions. 

> Then down the road we could consider putting both package and release into Docker.

yes - that's a good long term goal

Reply:
The environment set up is added to the deploy profile in commit [ca0cdf0](https://github.com/apache/tinkerpop/pull/1797/commits/ca0cdf047710895f38a143fd0b9eada3604620f6). There could be issues we won't realized until the actual release process though (hopefully not).